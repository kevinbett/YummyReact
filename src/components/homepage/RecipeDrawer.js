import React from 'react';
import Drawer from 'material-ui/Drawer';
import {List, ListItem} from 'material-ui/List';
import AppBar from 'material-ui/AppBar';
import RemoveRedEye from 'material-ui/svg-icons/image/remove-red-eye';
import Divider from 'material-ui/Divider';
import {orange700} from 'material-ui/styles/colors';
import {BrowserRouter} from "react-router-dom";

class RecipeDrawer extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        open: false
      };
    }
    handleViewcategories(event){
        event.preventDefault();
        window.location.assign('/dashboard/categories');
    }
  
    render() {
      return (
        <BrowserRouter>
        <div>
          <Drawer open={this.props.open} zDepth={0} containerStyle={{overflowX: 'hidden'}}>
          <AppBar title="Yummy Recipes" style={{backgroundColor: orange700}} showMenuIconButton={false} />
            <List>
            <ListItem key={7} primaryText="Back to categories" initiallyOpen={true}
                  onClick={this.handleViewcategories}
                  leftIcon={<RemoveRedEye />}
                />
            </List>
            <Divider />
          </Drawer>
        </div>
        </BrowserRouter>
      );
    }
  }
  export default RecipeDrawer;