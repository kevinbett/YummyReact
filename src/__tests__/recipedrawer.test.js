import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import RecipeDrawer from '../components/homepage/RecipeDrawer';

describe('RecipeDrawer Component', () => {
    const wrapper = shallow(<RecipeDrawer />);
    const preventDefault = jest.fn();
    it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('renders state initially', () => {
        expect(wrapper.state().username).toEqual('');
        expect(wrapper.state().password).toEqual('');
    })
});