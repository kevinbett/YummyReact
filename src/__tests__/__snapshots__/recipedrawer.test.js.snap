// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDrawer Component renders without crashing 1`] = `
<BrowserRouter>
  <div>
    <Drawer
      containerStyle={
        Object {
          "overflowX": "hidden",
        }
      }
      disableSwipeToOpen={false}
      docked={true}
      open={null}
      openSecondary={false}
      swipeAreaWidth={30}
      width={null}
      zDepth={0}
    >
      <AppBar
        showMenuIconButton={false}
        style={
          Object {
            "backgroundColor": "#f57c00",
          }
        }
        title="Yummy Recipes"
        zDepth={1}
      />
      <List>
        <ListItem
          autoGenerateNestedIndicator={true}
          containerElement="span"
          disableKeyboardFocus={false}
          disabled={false}
          initiallyOpen={true}
          insetChildren={false}
          key="7"
          leftIcon={<ImageRemoveRedEye />}
          nestedItems={Array []}
          nestedLevel={0}
          onClick={[Function]}
          onKeyboardFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onNestedListToggle={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          open={null}
          primaryText="Back to categories"
          primaryTogglesNestedList={false}
          secondaryTextLines={1}
        />
      </List>
      <Divider
        inset={false}
      />
    </Drawer>
  </div>
</BrowserRouter>
`;
